### PR TITLE
👔 add configurable default payment term for invoice sync

### PR DESCRIPTION
**Pull request by pedropote**
## Objectiu
Carregar per defecte un payment term configurable per enviar a Odoo quan les factures no ho tenen informat. És un camp obligatori.

## Targeta on es demana o Incidència
Al fer sync massiva ens hem trobat moltes factures que han donat error per tenir aquesta dada buida.

## Comportament antic
Passava el camp com False quan a la factura no tenia valor.

## Comportament nou
Passarà un valor configurat quan a la factura no tinga valor.

## Comprovacions

- [x] Hi ha testos
- [x] Reiniciar serveis
- [ ] Actualitzar mòdul
- [x] Script de migració
- [ ] Modifica traduccions

<details><summary><h3>Greptile Summary</h3></summary>

This PR adds a configurable default payment term (`odoo_default_erp_payment_term`) that is applied to invoices sent to Odoo when the invoice itself has no payment term set, fixing bulk-sync failures caused by the field being required by the Odoo API.

- Adds a `_get_default_odoo_payment_term_id` helper that reads an ERP-side ID from config, maps it to an Odoo ID via `get_odoo_id_by_erp_id`, and warns (and returns `False`) at each failure point — the three distinct warning paths cover: invalid config value, unconfigured (`0`), and missing Odoo mapping.
- Hooks the default into `hook_last_modifications` only when `invoice_payment_term_id` is absent/falsy, so existing payment terms are never overridden.
- Includes a migration script to inject the new config record and four new unit tests covering the happy path, missing mapping, unconfigured default, and the no-override case. Also fixes a pre-existing test bug where `result_data` was passed as input instead of `input_data`.
</details>

<details><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge. The change is well-scoped: it only fills in invoice_payment_term_id when the field is absent, never overrides an existing value, and the default config ships as 0 (effectively disabled) so no behaviour changes until an operator explicitly sets the value.

All three failure modes (invalid config, unconfigured, missing Odoo mapping) return False without raising and are covered by tests. The migration script correctly skips fresh installs and updating_all runs. No file shows an unchecked code path on the changed logic.

No files require special attention.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| som_sync_openerp/models/account_invoice.py | Adds _get_default_odoo_payment_term_id helper and wires it into hook_last_modifications; logic is correct and all edge cases are handled. |
| som_sync_openerp/tests/tests_account_invoice.py | Adds four new tests covering happy path, mapping failure, unconfigured default, and no-override; also fixes the pre-existing bug of passing result_data instead of input_data to hook_last_modifications. |
| som_sync_openerp/data/som_sync_openerp_data.xml | Adds the odoo_default_erp_payment_term config record with a safe default of 0 (disabled). |
| som_sync_openerp/migrations/5.0.26.6.0/post-0001_add_payment_term_config.py | Standard post-migration script that loads the updated XML; correctly short-circuits on fresh installs and updating_all mode. |

</details>

</details>

<details><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[hook_last_modifications called] --> B{move_type in_invoice / in_refund?}
    B -- Yes --> C[Set preferred_payment_method_line_id from config]
    B -- No --> D{invoice_payment_term_id falsy?}
    C --> D
    D -- No --> Z[Return data unchanged]
    D -- Yes --> E[_get_default_odoo_payment_term_id]
    E --> F{Read odoo_default_erp_payment_term from res.config}
    F --> G{Valid int?}
    G -- No --> H[WARNING: invalid config value - return False]
    G -- Yes --> I{value > 0?}
    I -- No --> J[WARNING: not configured - return False]
    I -- Yes --> K[get_odoo_id_by_erp_id account.payment.term]
    K --> L{Odoo ID found?}
    L -- No --> M[WARNING: no mapping - return False]
    L -- Yes --> N[Return odoo_payment_term_id]
    H --> O{default_payment_term_id truthy?}
    J --> O
    M --> O
    N --> O
    O -- Yes --> P[Set invoice_payment_term_id]
    O -- No --> Z
    P --> Z
```
</details>

<sub>Reviews (2): Last reviewed commit: ["🐛 finx flake 8"](https://github.com/som-energia/som_sync_openerp_modules/commit/4905cae356f5eaa0293000c53c117a03869a65b6) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=36919711)</sub>

### CODE REVIEW

File: som_sync_openerp/models/account_invoice.py [hunk 1]
```diff
@@ -337,13 +340,45 @@ def hook_last_modifications(self, cr, uid, data, context=None):
...
+        except (TypeError, ValueError):
+            logger.warning(
+                "Invalid odoo_default_erp_payment_term config value: %s",
+                erp_payment_term_id,
+            )
+            return False
+
+        if not erp_payment_term_id:
+            logger.warning("odoo_default_erp_payment_term is not configured")
+            return False
```
[the remark this answers is not in the document]
Comment (pedropote) [reply som_sync_openerp/models/account_invoice.py:373]: The thing is that with 0 value the invoice sync will fail also because the API requieres and int value greatter than 0. Considering that, do you think the same?

File: som_sync_openerp/migrations/5.0.25.5.0/post-0002_add_payment_term_config.py
Comment (oriolpiera) [inline som_sync_openerp/migrations/5.0.25.5.0/post-0002_add_payment_term_config.py:1]: @pedropote la carpeta del migració hauria de ser 5.0.25.5.0

**Review by oriolpiera — APPROVED** [review]
